openedx/features/job_board/helpers.py

```diff
@@ -14,3 +18,24 @@ def validate_file_size(file):
         raise ValidationError(
             _('File size must not exceed {size} MB').format(size=LOGO_IMAGE_MAX_SIZE / 1024 / 1024)
         )
+
+
+@deconstructible
+class UploadToPathAndRename(object):
+    """
+    Rename file uploaded by user.
+    """
+
+    def __init__(self, path):
+        self.sub_path = path
+
+    def __call__(self, instance, filename):
+        file_extension = filename.split('.')[-1] if filename else ''
+        # get filename
+        if instance.pk:
+            filename = 'logo_{}.{}'.format(instance.pk, file_extension)
+        else:
+            # set filename as random string
+            filename = 'logo_{}.{}'.format(uuid4().hex, file_extension)
+        # return the whole path to the file
```

Comment:
Let's please simplify this code to
```
unique_name = instance.pk or uuid4().hex
filename = 'logo_{}.{}'.format(unique_name, file_extension)
```

Reply:
Let's please replace logo with some generic name like image or file or of your choice. 
Or we can make it generic by checking either it is file or image
```
name_prefix = 'file' if filename in ['pdf'] else 'image'
unique_name = instance.pk or uuid4().hex
filename = '{}_{}.{}'.format(name_prefix, unique_name, file_extension)
```